src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,192 @@
+import { Component } from "eightbittr/lib/Component";
+import { ICallbackRegister, IMod } from "modattachr/lib/IModAttachr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { INewPokemon } from "../constants/Pokemon";
+
+/**
+ * Interface for holding items and their probability of being held.
+ */
+interface IItemProbabilities {
+    /**
+     * Name of them item.
+     */
+    name: string;
+
+    /**
+     * Probability of Pokemon having item.
+     */
+    probability: number;
+}
+
+ /**
+  * Mod that randomizes items found on wild Pokemon.
+  */
+export class RandomHeldItemsMod<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> implements IMod {
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         onRandomItems: (chosenInfo: INewPokemon): INewPokemon => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const generatedNumber: number = this.gameStarter.numberMaker.randomReal1();
+             let counter: number = 0;
+             for (const chosenObject of this.typeItems[pokemonType]) {
+                if (this.itemChooser(chosenInfo, counter, chosenObject, generatedNumber)) {
+                    break;
+                }
+             }
+
+             return chosenInfo;
+         }
+     };
+
+     /**
+      * Handles which item is chosen for onRandomItems.
+      */
+      private itemChooser(chosenInfo: INewPokemon, counter: number, chosenObject: IItemProbabilities, generatedNumber: number): boolean {
+            counter += chosenObject.probability;
```

Comment:
This doesn't work. Changing `counter` here won't modify the variable in the other function.